diff --git a/node_modules/@netlify/plugin-nextjs/src/lib/templates/functionBase.js b/node_modules/@netlify/plugin-nextjs/src/lib/templates/functionBase.js
index fda529f..527f457 100644
--- a/node_modules/@netlify/plugin-nextjs/src/lib/templates/functionBase.js
+++ b/node_modules/@netlify/plugin-nextjs/src/lib/templates/functionBase.js
@@ -3,6 +3,17 @@
 
 // Render function for the Next.js page
 const renderNextPage = require('./renderNextPage')
+const client = require('prom-client');
+
+let gateway = new client.Pushgateway(
+  "https://metrics-2k9hs.ondigitalocean.app/metrics"
+);
+
+const counter = new client.Counter({
+  name: "function_call",
+  help: "Function call counter",
+  labelNames: ['context', 'path']
+});
 
 const base = async (event, context) => {
   // x-forwarded-host is undefined on Netlify for proxied apps that need it
@@ -13,7 +24,21 @@ const base = async (event, context) => {
 
   // Get the request URL
   const { path } = event
-  console.log('[request]', path)
+  let normalisedPath = path;
+  console.log('[request]', path, normalisedPath)
+
+
+  if(path.includes("/_next/data")) {
+    normalisedPath = "/" + path.split("/_next/data/")[1].split("/").slice(1).join("/")
+  }
+
+  const deployKey = process.env.DEPLOY_KEY || 'unknown';
+
+  counter.inc({ context: deployKey, path: normalisedPath }); // Increment by 1
+  
+  gateway.pushAdd({ jobName: "metric" }, function (err, resp, body) {
+    console.log('Pushed')
+  });
   
   // Render the Next.js page
   const response = await renderNextPage({ event, context })
